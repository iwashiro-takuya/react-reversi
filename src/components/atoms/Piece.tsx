import React from 'react';
import styled from 'styled-components';
import Const from 'src/const';

import { PlayerValType } from 'src/types';

const { Color, PlayerVal } = Const;
interface PieceProp {
  playerVal: PlayerValType;
  onclick?: () => void;
  dataCy?: string;
}

const Piece: React.FC<PieceProp> = ({ playerVal, onclick, dataCy }) => {
  switch (playerVal) {
    case PlayerVal.WHITE:
      return <StyledPiece color={Color.PC_WHITE} />;
    case PlayerVal.BLACK:
      return <StyledPiece color={Color.PC_BLACK} />;
    default:
      return (
        <StyledPiece
          color={Color.PC_INVISIBLE}
          onClick={onclick}
          data-cy={dataCy}
        />
      );
  }
};

type StyledPieceProp = {
  color: string;
  onClick?: () => void;
};
const StyledPiece = styled.span<StyledPieceProp>`
  display: block;
  width: 80%;
  height: 80%;
  background-color: ${(props) => props.color};
  border-radius: 50%;

  &:hover {
    cursor: pointer;
  }
`;

export default Piece;
