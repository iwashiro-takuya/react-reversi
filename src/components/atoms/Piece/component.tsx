import React from 'react';
import Const from 'src/const';
import styled, { css } from 'styled-components';

const { Color, PlayerVal } = Const;

export interface PieceProp {
  playerVal: number;
  onclick?: () => void;
  dataCy?: string;
}
const Piece: React.FC<PieceProp> = ({ playerVal, onclick, dataCy }) => {
  switch (playerVal) {
    case PlayerVal.WHITE:
      return <StyledPiece color={Color.PC_WHITE} />;
    case PlayerVal.BLACK:
      return <StyledPiece color={Color.PC_BLACK} />;
    default:
      return (
        <StyledPiece
          color={Color.PC_INVISIBLE}
          onClick={onclick}
          data-cy={dataCy}
        />
      );
  }
};

type StyledPieceProp = {
  color: UnionValType<typeof Color>;
};
const StyledPiece = styled.span<StyledPieceProp>`
  display: block;
  width: 80%;
  height: 80%;
  background-color: ${(props) => props.color};
  ${(props) =>
    props.color === Color.PC_INVISIBLE
      ? css`
          border: none;
        `
      : css`
          border: 1px solid ${Color.BD_BLACK};
        `}

  border-radius: 50%;

  &:hover {
    cursor: pointer;
  }
`;

export default Piece;
