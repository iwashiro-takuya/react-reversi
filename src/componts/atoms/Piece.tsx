import React from 'react';
import styled from 'styled-components';
import Const from '../../const';

const { Color } = Const;

export default () => {
  return <Piece></Piece>;
};

const Piece = styled.span`
  &:hover {
    cursor: pointer;
  }
  background-color: ${Color.BG_BLACK};
  border-radius: 50%;
  display: block;
  height: 80%;
  width: 80%;
`;
