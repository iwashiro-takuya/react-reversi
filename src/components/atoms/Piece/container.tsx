import React from 'react';
import Const from 'src/const';
import Piece from 'src/components/atoms/Piece/component';

import { PlayerValType } from 'src/@types';

const { Color, PlayerVal } = Const;
interface PieceProp {
  playerVal: PlayerValType;
  onclick?: () => void;
  dataCy?: string;
}

const EnhancedPiece: React.FC<PieceProp> = ({ playerVal, onclick, dataCy }) => {
  switch (playerVal) {
    case PlayerVal.WHITE:
      return <Piece color={Color.PC_WHITE} />;
    case PlayerVal.BLACK:
      return <Piece color={Color.PC_BLACK} />;
    default:
      return (
        <Piece color={Color.PC_INVISIBLE} onclick={onclick} dataCy={dataCy} />
      );
  }
};

export default EnhancedPiece;
