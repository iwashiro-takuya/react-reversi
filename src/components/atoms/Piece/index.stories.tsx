import React from 'react';
import styled from 'styled-components';
import { Story } from '@storybook/react/types-6-0';
import Const from 'src/const';
import Piece, { PieceProp } from './component';

const { PlayerVal } = Const;

export default {
  title: 'Atoms/Piece',
  component: Piece,
  argTypes: {
    playerVal: {
      control: {
        type: 'select',
        options: {
          white: PlayerVal.WHITE,
          black: PlayerVal.BLACK,
          invisible: PlayerVal.NONE,
        },
      },
    },
  },
};

const Template: Story<PieceProp> = (args) => (
  <Base>
    <Piece {...args} />
  </Base>
);

export const normal = Template.bind({});

normal.args = {
  playerVal: PlayerVal.WHITE,
};

const Base = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 80px;
  height: 80px;
`;
