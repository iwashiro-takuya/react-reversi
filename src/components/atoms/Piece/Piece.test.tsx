import React from 'react';
import renderer from 'react-test-renderer';
import Const from 'src/const';
import Piece from 'src/components/atoms/Piece/container';

const { PlayerVal } = Const;

describe('Piece unit test', () => {
  it('snapshot(white)', () => {
    const component = renderer.create(<Piece playerVal={PlayerVal.WHITE} />);
    expect(component).toMatchSnapshot();
  });
  it('snapshot(black)', () => {
    const component = renderer.create(<Piece playerVal={PlayerVal.BLACK} />);
    expect(component).toMatchSnapshot();
  });
  it('snapshot(clickable)', () => {
    const component = renderer.create(
      <Piece playerVal={PlayerVal.NONE} dataCy="clickable" />,
    );
    expect(component).toMatchSnapshot();
  });
});
