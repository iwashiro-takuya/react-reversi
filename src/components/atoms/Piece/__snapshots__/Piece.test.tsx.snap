// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piece unit test snapshot testing(black) 1`] = `
<span
  className="sc-bdnylx JBiyk"
  color="#000"
/>
`;

exports[`Piece unit test snapshot testing(clickable) 1`] = `
<span
  className="sc-bdnylx dPcEsT"
  color="#38A34B"
  data-cy="clickable"
/>
`;

exports[`Piece unit test snapshot testing(white) 1`] = `
<span
  className="sc-bdnylx cJlQFq"
  color="#fff"
/>
`;

exports[`Piece unit test snapshot(black) 1`] = `
<span
  className="sc-bdnylx JBiyk"
  color="#000"
/>
`;

exports[`Piece unit test snapshot(clickable) 1`] = `
<span
  className="sc-bdnylx dPcEsT"
  color="#38A34B"
  data-cy="clickable"
/>
`;

exports[`Piece unit test snapshot(white) 1`] = `
<span
  className="sc-bdnylx cJlQFq"
  color="#fff"
/>
`;
